### PR TITLE
feat(setup): Python setup 명령어 구현 [#46]

### LINKED ISSUE #46: feat(setup): Python setup 명령어 구현
## 개요
대화형 설정 마법사 + 비대화형 옵션을 Python으로 구현. `src/cli/setup.py`.

## 배경
- 현재 setup.sh는 `read -p` 기반 완전 대화형으로 테스트 어려움
- agent/git 연동 강화, setup_done flag 도입 필요
- config.py (별도 이슈) 기반으로 동작

## 대화형 모드 (옵션 없이 실행)
1. **root** (BOJ_SOLUTION_ROOT): 현 위치 / 직접 입력. 경로 유효성 + 확인 질문
2. **언어**: java/python만 선택 가능. 나머지 미지원 안내
3. **agent**: 선택지 (claude, antigravity, opencode, cursor, copilot, 기타)
   - agent별 실행 명령어 자동 매핑
   - 필요 시 설치 안내 (예: cursor CLI install)
   - 없으면 사용 불가 안내 + antigravity 무료 추천
4. **git 연동**:
   - 옵션: 이미 연동됨 / repo URL clone / gh create repo (gh 필수)
   - git config --global user.name, user.email 설정
   - push 인증은 기존 전역 설정 가정
5. **username**: BOJ userId 입력
6. **에디터**: 실행 명령어 (없으면 boj open 불가 안내). boj make 시 자동 boj open
7. **완료**: `~/.config/boj/setup_done` 생성
8. **사용법 출력**: 전 명령어(setup/make/open/run/commit/submit/review) 소개
9. **설정법 출력**: setup 옵션 소개

## 비대화형 옵션
- (없음): 전체 대화형 재설정
- `--check`: config 조회 (`boj_check_config`) 후 종료
- `--root [path]`: root 설정 (경로 검증 + "정말 설정?" 확인)
- `--lang`, `--username`, `--editor`, `--agent`: 개별 키 재설정

- **boj commit**: push 실패 시 "push 권한이 없습니다. `git config credential.helper` 또는 SSH 키를 설정하세요." 안내
- **boj commit**: username이 맞지 않거나 존재하지 않을 때 예외처리
- **모든 명령어**: setup_done 체크 로직은 각 명령어 Python 전환 시 적용 (setup 자체가 아닌 호출하는 쪽에서 체크)

## 파일
- 생성: `src/cli/setup.py`, `tests/unit/test_setup.py`
- 의존: `src/core/config.py` (config 이슈)

## 진행 방식 (TDD)
1. test-strategy + edge-cases 문서 작성
2. 컨펌 후 테스트 파일 구현 (Red)
3. 로직 구현 (Green)
4. 문서 업데이트: COMMAND-SPEC.md, user-guide.md, DEVLOG.md

## 의존성
- config 모듈 이슈 완료 필요
- 후속: install 이슈가 이 명령어에 의존

### PR DESCRIPTION
**Pull request by sueelly**
## Related Issue
Closes #46

## Type of Change
- [x] feat (새 기능)

## Summary

`src/commands/setup.sh` (284줄 Bash)를 Python `src/cli/boj_setup.py`로 재구현.
`src/core/config.py` (#45) 기반으로 동작하며, `prompter` 의존성 주입 패턴으로 대화형 로직을 100% 단위 테스트 가능하게 설계.

### 새 파일
- `src/cli/__init__.py` — cli 패키지
- `src/cli/boj_setup.py` — Python setup 명령어 (600줄)
- `tests/unit/test_setup.py` — 단위 테스트 (93개)
- `tests/integration/test_boj_setup_py.py` — 통합 테스트 (13개, subprocess 호출)

### 문서 업데이트
- `docs/COMMAND-SPEC.md` — boj setup 섹션 Python 신규 구현 추가, CLI→config 키 매핑 테이블
- `docs/user-guide.md` — setup 사용법 Python 기준으로 업데이트
- `docs/dev/test-strategy.md` — Section 14 추가 (Python setup 테스트 전략, S1~S15 커버리지 매핑)
- `docs/edge-cases.md` — S6 동작 수정 (에러 종료 → fallback), S11~S15 신규 케이스 추가
- `docs/records/DEVLOG.md` — [2026-03-12] #46 구현 기록

## Key Design Decisions

**대화형 6단계**: root → lang → agent → git → username → editor
- 기존 5단계에서 agent 단계 추가, 세션 쿠키 로직 제외 (별도 이슈)

**비대화형 옵션**: `--check`, `--root`, `--lang`, `--username`, `--editor`, `--agent`
- `--agent claude` → `AGENT_COMMANDS` 자동 매핑 (`claude -p --`)
- `--agent "custom-cmd"` → 그대로 저장

**git 연동 (step 4)**:
- 이미 연동 / repo URL clone / gh repo create 3가지 선택
- gh 미설치 시 설치 안내 + 다른 옵션으로 fallback

**agent 없음**: gemini 무료 추천 안내

**setup 완료**: `~/.config/boj/setup_done` 플래그 생성 + 전체 명령어 사용법 출력

## Test Evidence

```
tests/unit/test_setup.py         — 93 passed
tests/unit/test_config.py        — 31 passed (회귀 없음)
tests/integration/test_boj_setup_py.py — 13 passed
Total: 106 passed in 1.86s
```

**통합 테스트 커버 (IP1~IP11)**:
- `--check` 상태 출력, 미설정 표시
- `--lang` 유효/무효 검증 및 저장
- `--root` 유효/무효 경로 처리
- `--username`, `--editor` 저장
- `--agent` known(자동 매핑) / custom(직접 저장)
- 복수 옵션 동시 적용
- 저장 후 `--check` 반영 확인

## Checklist
- [x] 커밋 메시지 conventional commits 형식
- [x] 테스트 통과 (106 passed)
- [x] 하드코딩 시크릿 없음
- [x] TODO/FIXME 없음 (세션 쿠키는 별도 이슈로 명시)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Introduces a new Python `boj setup` implementation that writes config, invokes git/gh subprocesses, and changes user-facing setup behavior; regressions could impact initial configuration/onboarding paths.
> 
> **Overview**
> Replaces the *behavioral spec* of `boj setup` with a new Python implementation (`src/cli/boj_setup.py`) that supports `--check` plus non-interactive setters for root/lang/username/editor/agent, and an interactive 6-step wizard (root → lang → agent → git/gh repo linkage → username → editor) that marks `setup_done` and prints a usage guide.
> 
> Adds comprehensive pytest coverage: unit tests for argument parsing, interactive flows, edge cases (e.g., Ctrl+C exit 130, unsupported language retry, gh-missing fallback), and subprocess-based integration tests to verify config files are persisted and reflected by `--check`. Documentation is updated to describe the new Python setup flow, CLI→config key mapping, expanded edge-case matrix (S1–S15), and the associated test strategy/devlog entries.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit e36c5891bbb9da3fc122056eaa06da8855c1f1ef. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/cli/boj_setup.py [hunk 1]
```diff
@@ -0,0 +1,600 @@
...
+        if not change.strip().lower().startswith("y"):
+            print(f"  유지: {current}")
+            return current
+
+    print(f"  에디터 실행 명령어를 입력하세요.")
+    print(f"  예: code (VS Code), vim, nano, idea")
+    editor = prompter(f"  에디터 [code]: ").strip()
+    if not editor:
+        editor = ""
+        print(f"  {YELLOW}에디터 미설정. boj open 명령어를 사용할 수 없습니다.{NC}")
```
[the remark this answers is not in the document]
Comment (sueelly) [reply src/cli/boj_setup.py]: onfig_get (78–105행)

환경변수: BOJ_EDITOR 등이 있으면 그걸 쓰고, 값이 비어 있거나 공백만 있으면 다음 단계로 넘어감.
파일: ~/.config/boj/editor를 읽을 때
val = config_file.read_text().strip() 후 if val: return val
→ 값이 빈 문자열이면 if val이 False라서 return하지 않고 마지막의 return default 로 간다.
즉 editor가 파일에 빈 값으로 저장돼 있거나, 키가 없으면
호출 시 넘긴 default를 그대로 반환한다.
config_get("editor", "code") → "code"
config_get("editor", "") → ""
DEFAULTS["editor"]는 "code"이고, get_effective_config()(241행)에서는
editor = config_get("editor", DEFAULTS["editor"]) 로 불러서, **실제 사용할 때는 editor가 비어 있으면 "code"**가 나옵니다.
